lib/ansible/modules/cloud/amazon/ec2_eni.py

```diff
@@ -475,26 +473,30 @@ def detach_eni(eni, module):
         module.exit_json(changed=False, interface=get_eni_info(eni))
 
 
-def find_eni(connection, module):
+def uniquely_find_eni(connection, module):
 
     eni_id = module.params.get("eni_id")
-    subnet_id = module.params.get('subnet_id')
     private_ip_address = module.params.get('private_ip_address')
     instance_id = module.params.get('instance_id')
     device_index = module.params.get('device_index')
 
     try:
         filters = {}
-        if subnet_id:
-            filters['subnet-id'] = subnet_id
+
+        # proceed only if we're univocally specifying an ENI
+        if eni_id is None and private_ip_address is None and (instance_id is None and device_index is None):
```

Comment:
Since a lot of people have been relying on broken functionality (filtering without specifying an eni_id, for instance), perhaps we want to add something to the docs, clarifying how modification should work (and that subnet_id isn't used to filter). Or perhaps we don't want to add to the docs and instead should just let people know when relevant issues are run across that the functionality only ever would have worked due to chance.